benchmarks/melt_column_differenciation/melt_SramekRicardBercovici2007.cc

```diff
@@ -0,0 +1,703 @@
+         */
+        virtual void melt_fractions (const MaterialModel::MaterialModelInputs<dim> &in,
+                                     std::vector<double> &melt_fractions) const;
+
+        /**
+         * @name Reference quantities
+         * @{
+         */
+        virtual double reference_viscosity () const;
+
+        virtual double reference_darcy_coefficient () const;
+
+
+        /**
+         * @}
+         */
+
+        /**
+         * @name Functions used in dealing with run-time parameters
+         * @{
+         */
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+        /**
+         * @}
+         */
+
+        virtual
+        void
+        create_additional_named_outputs (MaterialModel::MaterialModelOutputs<dim> &out) const;
+
+
+      private:
+        double reference_rho_s;
+        double reference_rho_f;
+        double reference_T;
+        double eta_0;
+        double xi_0;
+        double eta_f;
+        double thermal_viscosity_exponent;
+        double thermal_bulk_viscosity_exponent;
+        double thermal_expansivity;
+        double reference_specific_heat;
+        double thermal_conductivity;
+        double reference_permeability;
+        double alpha_phi;
+        double depletion_density_change;
+        double depletion_solidus_change;
+        double pressure_solidus_change;
+        double surface_solidus;
+        double compressibility;
+        double melt_compressibility;
+        bool include_melting_and_freezing;
+        double melting_time_scale;
+        double alpha_depletion;
+        double delta_eta_depletion_max;
+
+        // entropy change upon melting
+        double peridotite_melting_entropy_change;
+
+        virtual
+        double
+        melt_fraction (const double temperature,
+                       const double pressure,
+                       const double depletion) const;
+    };
+
+  }
+}
+
+
+
+/*
+  Copyright (C) 2015 - 2019 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
```

Comment:
Please remove this copyright notice. Each file in ASPECT requires exactly one of these notices at the very top of the file.